src/components/Carousel/Carousel.tsx

```diff
@@ -192,7 +198,11 @@ export const Carousel: FC<CarouselProps> = ({
               {leftControl ? leftControl : <DefaultLeftControl theme={customTheme} />}
             </button>
           </div>
-          <div className={theme.root.rightControl}>
+          <div
+            className={`${theme.root.rightControl} transition-opacity duration-300 ${
+              activeItem === items.length - 1 && autoHideControls ? 'pointer-events-none opacity-0' : 'opacity-100'
+            }`}
+          >
```

Comment:
Thanks for your contribution.

Would opacity be the best approach in this case? W/opacity the controls will remain available to user interactions on the dom, while display:none would prevent, that.

Reply:
Good catch, yes, that's right.

Reply:
Hello. If with controls remaining available you mean that the user will go from last image to first image even with control hided, in my tests when control hides click has no effect. In this video you will see a circle in the pointer when I click:

https://github.com/themesberg/flowbite-react/assets/36115659/49c60eb2-3795-4482-887b-2dfbb117bba5

Please correct me if I misunderstood.

Reply:
I would still change this to `display:none` because this may come up as still clickable for users navigating with Tab.

I can't confirm that since you didn't add an example to the docs, which, if you could, would be great.

With that said, I'm still not sure if we can merge this because this is a UX feature that isn't available in the vanilla library at all.

Reply:
You are right! Thanks for your feedback I will implement this in my forked repo. Please let me know if you'd like this feature here so I can update the PR.

Reply:
Keep in mind that if the navigation button/icon disappears from the DOM, when the user spam clicks next (or prev) until he gets to the last, if in the future we'll implement click-to-fullscreen (or similar), then this behaviour would be very unpredictable.

So the case study is, the next/prev icons are wrapped in a `button` element in order to be focusable/accessible and the button becomes `disabled=true` when is last or first page, according to the direction, this way tabbing is not possible (CC @tulup-conner ) and also clicking the disabled button should stop event propagation.

#just_thoughts 🤔